doc/api-migration.md

```diff
@@ -1,4 +1,85 @@
-## Migrating to the new API
+# Migrating to new APIs
+
```

Comment:
Should we have some preamble which explains how each section will explain how to migrate from a given version to the next?